### PR TITLE
Fix intermittent loading image errors in NFT thumbnails

### PR DESCRIPTION
**Pull request by jxom**
Fixes TEAM2-42

## What changed (plus any additional context for devs)

This PR fixes an issue that has been existant for quite a while when rapidly scrolling through large collections. 

The issue is that some NFT thumbnails throw an error upon fetching for the image, and as a result, displays the fallback image component (monospace text). It seems that the `SDWebImage` native dependency was throwing a "Operation cancelled by user during sending the request" error in our `ImgixImage` component for a few NFTs. I'm not really sure why this was happening, but I don't really have the capacity to learn Objective C and dig through `SDWebImage` to find the true culprit of why this error is throwing. For now, the proposed fix is to remount the erroneous image until it loads successfully. 

## PoW (screenshots / screen recordings)

Before:

https://user-images.githubusercontent.com/7336481/176613625-4a2eb384-26fc-44b7-b37e-abcb4a39505d.mp4

After:

https://user-images.githubusercontent.com/7336481/176613376-1f14905d-79a2-4412-9bdb-756b0bbe0e19.mp4

## Dev checklist for QA: what to test

- Rapidly scroll through large NFT collections and ensure all the thumbnails resolve successfully.

### CODE REVIEW

**Comment by skylarbarrera** [comment]
im fine with this change as a temp fix but we should create another ticket and maybe assign to performance so they can dig a little deeper


no images are loading for me locally but ill wait for QA to confirm: https://cloud.skylarbarrera.com/Screen-Recording-2022-07-01-11-07-01.mp4

**Review by BrodyHughes — APPROVED** [review]
Looks amazing. great fix! @skylarbarrera local build for me looks good.

**Comment by BrodyHughes** [comment]
https://user-images.githubusercontent.com/41711440/176949716-4320289a-7f2f-4bcf-96f8-ac0ec32d29a4.mov

**Comment by BrodyHughes** [comment]
It is struggling a lot performance wise on larger wallets tho (vitalik.eth), so maybe another ticket for that like Skylar is saying.

**Comment by BrodyHughes** [comment]
https://user-images.githubusercontent.com/41711440/176951701-9e9b26f8-2d63-4087-8aba-1dadc2610283.mp4

File: src/components/images/ImgixImage.tsx [hunk 1]
```diff
@@ -41,12 +44,31 @@ class ImgixImage extends React.PureComponent<
...
+        {...props}
+        key={retryCount}
+        onError={(err: any) => {
+          if (retryOnError && retryCount < maxRetries) {
+            this.setState(({ retryCount }) => ({ retryCount: retryCount + 1 }));
+          } else {
+            // @ts-expect-error
+            props.onError?.(err);
+          }
+        }}
```
Comment (terrysahaidak) [inline src/components/images/ImgixImage.tsx:68]: Since it's a class component, can we move this to the method

**Review by terrysahaidak — CHANGES_REQUESTED** [review]
I think it will introduce performance degradation for coin icons since we don't handle any 404 which is really common for coin icons. also when we have network error - it will basically bomb the entire screen with these events too, so i'm not sure if this is really a good solution for this particular problem.

**Review by osdnk — CHANGES_REQUESTED** [review]
I agree with Terry, we should check for 404

Retrying is also limiting the problem and we pay with the performance

**Comment by jxom** [comment]
@terrysahaidak @osdnk – Good catch! I have added bindings to the ObjC implementation to pass through the message/status code, and not run the retry stuff if there is a 404. 

In terms of performance degradation, this shouldn't affect coin icons as the `retryOnError` flag is only truthy for the `UniqueTokenImage` component. These components also error very rarely (only when scrolling large lists) and even then, only a _few_ of them error, but there is defs a better solution to this. Perhaps this could be a temp fix, with a long-term solution in mind?

**Comment by jxom** [comment]
> It is struggling a lot performance wise on larger wallets tho (vitalik.eth), so maybe another ticket for that like Skylar is saying.

@BrodyHughes – this might be an existing issue, i think this issue was happening on iOS only

**Review by osdnk — APPROVED** [review]

**Review by terrysahaidak — APPROVED** [review]

**Pull request merged**